src/core/socialprovider.js

```diff
@@ -167,7 +175,6 @@ XMPPSocialProvider.prototype.connect = function(continuation) {
 
     if (this.client) {
       this.client.end();
-      delete this.client;
```

Comment:
My understanding of this code is that calling ```this.client.end()``` will result in in an "end" event being fire which results in logout being called (lines 195-208).  The reason why ```delete this.client``` was removed is because that "end" callback could be invoked asynchronously, i.e. first we would call end(), then we would delete this.client, then the "end" callback would be invoked but because this.client wasn't set nothing would happen and the user would remain logged in but with this.client==null, leading to bugs.

Possibly we could just change the code to call logout() here rather than this.client.end()

And yes ideally we should do more to parse the errors and ignore the ones that aren't fatal or otherwise recover gracefully.  But I think until we know more it's best to log the user out in case of errors - otherwise they risk being logged in but nothing working.  Plus we have and will be improving automatic reconnect in the uProxy code so this will be less of a UX issue